### DISCUSSION TITLE
I love this and finally got it set up how I want. I need a backup

### DISCUSSION THREAD

**Discussion by TimVanDyke** [regolith-linux/regolith-desktop#867] [Q&A] [answered]
Hello, I LOVE Regolith. This is my most recent attempt at a Linux desktop and I really really think this is it. 

I've tried many times in the past but it didn't quite scratch the tiling window manager itch I had and never looked good when I attempted to set one up myself or migrate away from XFCE myself for example. 

I LOVE this. I can't say it enough. 

The problem is I never know when to leave well enough alone. I have everything (even photo editing and photo printing) working GREAT (except bluetooth but I assume that's just one afternoon away and I almost never need it anyway). 

I don't want to break my current setup. I've done my best to document what I've done and remember things but honestly I did a pretty bad job. I know I don't remember everything I've done. 

Is there some way to make an ISO file or a system backup or something? 

I am sure there is I just didn't see the right way on the desktop and am afraid to make more changes for fear of destroying the setup I have: 

Here's what I remember doing (the docs were super helpful)
 - Gruvbox theme
 - minor keyboard shortcut tweaks
 - minor look and feel tweaks
 - stable undervolt starting on boot
 - all my apps installed and logged in
 - minor bug fixes (like making it so settings opens from ilia with turning off the previous pane or whatever it was)
 - disable nvidia gpu (and reneable when I actually need it with prime-select)
 - remapped caps -> esc using /usr/share/X11/xkb/symbols/pc

It took me probably a few days to get it like this and a fair amount of reading. 

Left to do after I make a backup to make sure I don't destroy everything: 
 - LightDM gruvbox theme maybe (for sure some theme or something
 - fix bluetooth (eventually)

But I am getting to the point that I like it and it's stable enough now that starting fresh is a pretty bad risk to take at this point. 

TL;DR;

How can I make a backup I can restore from when I inevitably mess up my environment? (Sorry if this is a duplicate I searched the website and discussions for "backup" "restore" and didn't find anything)

**Comment by cfsmp3** [comment, marked as answer]
I recommend you create a private github repository and just use git.

You will have a safe backup plus version control.

Usually it would have much more than just regolith, but you can start with it.

Look up "using github to save dotfiles" and you'll find lots of literature.

My dotfiles repo contains regolith, neovim, alacritty and a few other things.

**Reply by TimVanDyke** [reply]
So that will save dotfiles but I've also added a script to /etc/init.d/ . Is that bad practice? Is there a good way to back that kind of change up? 


What about installed programs I installed via command line I don't even remember? Is there some sort of machine snapshot I can use?

**Reply by cfsmp3** [reply]
> So that will save dotfiles but I've also added a script to /etc/init.d/ . Is that bad practice? Is there a good way to back that kind of change up? 

That's probably not part of regolith, but I'd say same thing. It doesn't really belong in dotfiles but no one will stop you 😁

Combine this repo with GNU stow which will magically symlink you repo directory with the locations in the file system where stuff needs to go. 

Also, in your private repo, you can have a wiki, issue tracking and so on.
